src/main/java/org/zalando/nakadi/service/FloodService.java

```diff
@@ -0,0 +1,177 @@
+package org.zalando.nakadi.service;
+
+import org.apache.commons.collections.map.HashedMap;
+import org.apache.curator.framework.recipes.cache.ChildData;
+import org.apache.curator.framework.recipes.cache.TreeCache;
+import org.apache.zookeeper.KeeperException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.zalando.nakadi.domain.EventType;
+import org.zalando.nakadi.exceptions.NakadiException;
+import org.zalando.nakadi.repository.db.EventTypeCache;
+import org.zalando.nakadi.repository.zookeeper.ZooKeeperHolder;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+@Component
+public class FloodService {
```

Comment:
I agree I think it could be even inappropriate to use zk for such loads but this task is about to have a way to block an aggressive producer/consumer constantly until team will decide to unblock it. This feature was introduced due to the last experiences on live as a fast solution.